docs/dev-tutorial.md

```diff
@@ -0,0 +1,247 @@
+# Development Tutorial
+
+## Before you contribute
+
+First, you need to fork your own repository.
+
+If you already have a good idea: you can propose it in an issue, explaining your thoughts and seeing if other contributors have any suggestions before starting to write code.
+
+If you don't have a specific idea: you can browse the issues and look for something that interests you. For beginners, especially pay attention to issues labeled "good first issue."
+
+## Install Antrea on kind
+
+Below, we will explain how to set up an Antrea development environment on Linux using Kind. First, you need to ensure that you have installed the following dependencies: go, docker, kubectl, and kind.
+
+Next, clone the Antrea code:
+
+```
+git clone git@github.com:$YOUR_GITHUB_USERNAME/antrea.git
+```
+
+You can now run the unit tests:
+
+```
+make test-unit
+```
+
+In theory, all unit tests should pass.
+
+Next, try building the local Docker image:
+
+```
+make antrea-agent antrea-controller antrea-cni antctl-release
+NO_PULL=1 make ubuntu
+```
```

Comment:
I don't have a macOS environment, but `make ubuntu` is for building an Antrea docker image based on ubuntu docker image, so I think it's OK for the macOS?

If this doesn't work, could you tell me how to build an image on macOS?

Reply:
You are mostly right. `make ubuntu` copies the binaries under the local `bin/` directory to an image. It can be quite fast and convenient, but the binaries that you build need to be built for the correct runtime platform.
`make antrea-agent antrea-controller antrea-cni antctl-release` will produce 3 linux binaries and one darwin binary (`antctl`), so using `make ubuntu` in that case is not technically correct, as it will copy a darwin binary to a linux container image.
 
The safest target is `make build-ubuntu` (or `make build`, or `make all`, or simply `make`, they all mean the same thing). It takes longer to run typically, but is easier to use and is more universal (as the Antrea binaries are built as part of the docker build process).

Reply:
In my computer `make` is much slower than `make ubuntu`. I think it's because `make` can't make use of the cache of `go build`. So I prefer using `make ubuntu`.

Then there comes a problem: `make antctl-linux` saves the output binary in `bin/antctl-linux`, not `bin/antctl`, so if I use `make antctl-linux` instead of `make antctl-release`, the built image won't have `antctl` command.

I think we can add a new Makefile target called `ubuntu-local` (or some nicer name. I have difficulty naming, to be honest), which is equivalent to `make` but doesn't build in a docker image?

Reply:
I now recall that `make bin` builds all the binaries for the Linux OS:
```
$ rm -rf bin/
$ make bin
GOOS=linux go build -o /Users/abas/vmware/antrea/bin  -ldflags ' -X antrea.io/antrea/pkg/version.Version=v1.14.0-dev -X antrea.io/antrea/pkg/version.GitSHA=1f3c549f3 -X antrea.io/antrea/pkg/version.GitTreeState=dirty -X antrea.io/antrea/pkg/version.ReleaseStatus=unreleased' antrea.io/antrea/cmd/...
$ ls -l bin/
total 905560
-rwxr-xr-x  1 abas  staff  82637141 Aug 16 15:26 antctl
-rwxr-xr-x  1 abas  staff  97331975 Aug 16 15:26 antrea-agent
-rwxr-xr-x  1 abas  staff  75666010 Aug 16 15:26 antrea-agent-simulator
-rwxr-xr-x  1 abas  staff  26713269 Aug 16 15:25 antrea-cni
-rwxr-xr-x  1 abas  staff  89570511 Aug 16 15:26 antrea-controller
-rwxr-xr-x  1 abas  staff  91713962 Aug 16 15:26 flow-aggregator
```

So there should be no need for another Makefile target.

If you want, you can improve `build/images/Dockerfile.ubuntu` so it only copies the necessary binaries (`antctl`, `antrea-agent`, `antrea-cni`, `antrea-controller`) to the container image, instead of all the binaries under `bin/`.